<!DOCTYPE html>
<html lang="zh-Hans">
<head>
  <meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=2">
<meta name="theme-color" content="#222">
<meta name="generator" content="Hexo 6.0.0">
  <link rel="apple-touch-icon" sizes="180x180" href="/images/apple-touch-icon-next.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/images/favicon-32x32-next.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/images/favicon-16x16-next.png">
  <link rel="mask-icon" href="/images/logo.svg" color="#222">

<link rel="stylesheet" href="/css/main.css">


<link rel="stylesheet" href="/lib/font-awesome/css/all.min.css">

<script id="hexo-configurations">
    var NexT = window.NexT || {};
    var CONFIG = {"hostname":"example.com","root":"/","scheme":"Pisces","version":"7.8.0","exturl":false,"sidebar":{"position":"left","display":"post","padding":18,"offset":12,"onmobile":false},"copycode":{"enable":true,"show_result":false,"style":"mac"},"back2top":{"enable":true,"sidebar":false,"scrollpercent":true},"bookmark":{"enable":false,"color":"#222","save":"auto"},"fancybox":false,"mediumzoom":false,"lazyload":false,"pangu":false,"comments":{"style":"tabs","active":null,"storage":true,"lazyload":false,"nav":null},"algolia":{"hits":{"per_page":10},"labels":{"input_placeholder":"Search for Posts","hits_empty":"We didn't find any results for the search: ${query}","hits_stats":"${hits} results found in ${time} ms"}},"localsearch":{"enable":false,"trigger":"auto","top_n_per_article":1,"unescape":false,"preload":false},"motion":{"enable":true,"async":false,"transition":{"post_block":"fadeIn","post_header":"slideDownIn","post_body":"slideDownIn","coll_header":"slideLeftIn","sidebar":"slideUpIn"}}};
  </script>

  <meta name="description" content="宏基因测序软件使用记录">
<meta property="og:type" content="article">
<meta property="og:title" content="宏基因组测序及分析流程整理（更新中）">
<meta property="og:url" content="http://example.com/2023/01/04/%E4%BA%8C%E4%BB%A3%E3%80%81%E4%B8%89%E4%BB%A3%E5%AE%8F%E5%9F%BA%E5%9B%A0%E7%BB%84%E6%B5%8B%E5%BA%8F%E5%8F%8A%E5%88%86%E6%9E%90%E6%B5%81%E7%A8%8B/index.html">
<meta property="og:site_name" content="ZEK&amp;YL&#39;s blog">
<meta property="og:description" content="宏基因测序软件使用记录">
<meta property="og:locale">
<meta property="og:image" content="https://raw.githubusercontent.com/Zhang-EK/blog_img/main/image.png">
<meta property="og:image" content="https://raw.githubusercontent.com/Zhang-EK/blog_img/main/111.png">
<meta property="og:image" content="https://raw.githubusercontent.com/Zhang-EK/blog_img/main/image2.png">
<meta property="og:image" content="https://raw.githubusercontent.com/Zhang-EK/blog_img/main/image%20(1).png">
<meta property="og:image" content="https://raw.githubusercontent.com/Zhang-EK/blog_img/main/32535.png">
<meta property="og:image" content="https://raw.githubusercontent.com/Zhang-EK/blog_img/main/4375.png">
<meta property="og:image" content="https://raw.githubusercontent.com/Zhang-EK/blog_img/main/452415.png">
<meta property="article:published_time" content="2023-01-04T00:00:00.000Z">
<meta property="article:modified_time" content="2023-01-12T16:54:02.916Z">
<meta property="article:author" content="ZEK YL">
<meta property="article:tag" content="生信软件学习笔记">
<meta name="twitter:card" content="summary">
<meta name="twitter:image" content="https://raw.githubusercontent.com/Zhang-EK/blog_img/main/image.png">

<link rel="canonical" href="http://example.com/2023/01/04/%E4%BA%8C%E4%BB%A3%E3%80%81%E4%B8%89%E4%BB%A3%E5%AE%8F%E5%9F%BA%E5%9B%A0%E7%BB%84%E6%B5%8B%E5%BA%8F%E5%8F%8A%E5%88%86%E6%9E%90%E6%B5%81%E7%A8%8B/">


<script id="page-configurations">
  // https://hexo.io/docs/variables.html
  CONFIG.page = {
    sidebar: "",
    isHome : false,
    isPost : true,
    lang   : 'zh-Hans'
  };
</script>

  <title>宏基因组测序及分析流程整理（更新中） | ZEK&YL's blog</title>
  






  <noscript>
  <style>
  .use-motion .brand,
  .use-motion .menu-item,
  .sidebar-inner,
  .use-motion .post-block,
  .use-motion .pagination,
  .use-motion .comments,
  .use-motion .post-header,
  .use-motion .post-body,
  .use-motion .collection-header { opacity: initial; }

  .use-motion .site-title,
  .use-motion .site-subtitle {
    opacity: initial;
    top: initial;
  }

  .use-motion .logo-line-before i { left: initial; }
  .use-motion .logo-line-after i { right: initial; }
  </style>
</noscript>

<link rel="alternate" href="/atom.xml" title="ZEK&YL's blog" type="application/atom+xml">
</head>

<body itemscope itemtype="http://schema.org/WebPage">
  <div class="container use-motion">
    <div class="headband"></div>

    <header class="header" itemscope itemtype="http://schema.org/WPHeader">
      <div class="header-inner"><div class="site-brand-container">
  <div class="site-nav-toggle">
    <div class="toggle" aria-label="Toggle navigation bar">
      <span class="toggle-line toggle-line-first"></span>
      <span class="toggle-line toggle-line-middle"></span>
      <span class="toggle-line toggle-line-last"></span>
    </div>
  </div>

  <div class="site-meta">

    <a href="/" class="brand" rel="start">
      <span class="logo-line-before"><i></i></span>
      <h1 class="site-title">ZEK&YL's blog</h1>
      <span class="logo-line-after"><i></i></span>
    </a>
  </div>

  <div class="site-nav-right">
    <div class="toggle popup-trigger">
    </div>
  </div>
</div>




<nav class="site-nav">
  <ul id="menu" class="main-menu menu">
        <li class="menu-item menu-item-主页">

    <a href="/" rel="section"><i class="fa fa-home fa-fw"></i>主页</a>

  </li>
        <li class="menu-item menu-item-标签">

    <a href="/tags/" rel="section"><i class="fa fa-tags fa-fw"></i>标签</a>

  </li>
        <li class="menu-item menu-item-分类">

    <a href="/categories/" rel="section"><i class="fa fa-th fa-fw"></i>分类</a>

  </li>
        <li class="menu-item menu-item-历史">

    <a href="/archives/" rel="section"><i class="fa fa-archive fa-fw"></i>历史</a>

  </li>
        <li class="menu-item menu-item-关于我们">

    <a href="/about/" rel="section"><i class="fa fa-user fa-fw"></i>关于我们</a>

  </li>
  </ul>
</nav>




</div>
    </header>

    
  <div class="back-to-top">
    <i class="fa fa-arrow-up"></i>
    <span>0%</span>
  </div>


    <main class="main">
      <div class="main-inner">
        <div class="content-wrap">
          

          <div class="content post posts-expand">
            

    
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="zh-Hans">
    <link itemprop="mainEntityOfPage" href="http://example.com/2023/01/04/%E4%BA%8C%E4%BB%A3%E3%80%81%E4%B8%89%E4%BB%A3%E5%AE%8F%E5%9F%BA%E5%9B%A0%E7%BB%84%E6%B5%8B%E5%BA%8F%E5%8F%8A%E5%88%86%E6%9E%90%E6%B5%81%E7%A8%8B/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="https://raw.githubusercontent.com/Zhang-EK/blog_img/main/WechatIMG22.jpeg">
      <meta itemprop="name" content="ZEK YL">
      <meta itemprop="description" content="2019.10.16">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="ZEK&YL's blog">
    </span>
      <header class="post-header">
        <h1 class="post-title" itemprop="name headline">
          宏基因组测序及分析流程整理（更新中）
        </h1>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">Posted on</span>

              <time title="Created: 2023-01-04 00:00:00" itemprop="dateCreated datePublished" datetime="2023-01-04T00:00:00+00:00">2023-01-04</time>
            </span>
              <span class="post-meta-item">
                <span class="post-meta-item-icon">
                  <i class="far fa-calendar-check"></i>
                </span>
                <span class="post-meta-item-text">Edited on</span>
                <time title="Modified: 2023-01-12 16:54:02" itemprop="dateModified" datetime="2023-01-12T16:54:02+00:00">2023-01-12</time>
              </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-folder"></i>
              </span>
              <span class="post-meta-item-text">In</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/2023/" itemprop="url" rel="index"><span itemprop="name">2023</span></a>
                </span>
            </span>

          
            <div class="post-description">宏基因测序软件使用记录</div>

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
        <h1 id="宏基因组测序及分析流程"><a href="#宏基因组测序及分析流程" class="headerlink" title="宏基因组测序及分析流程"></a>宏基因组测序及分析流程</h1><h2 id="二代测序数据拼接"><a href="#二代测序数据拼接" class="headerlink" title="二代测序数据拼接"></a>二代测序数据拼接</h2><p>质控完成之后，宏基因组二代测序的数据拼接软件使用基本和单菌拼接保持一致，目前大部分软件都支持宏基因组的拼接模式。</p>
<h3 id="SPAdes"><a href="#SPAdes" class="headerlink" title="SPAdes"></a>SPAdes</h3><p>该软件支持宏基因组的拼接模式，<strong>只需要加上参数–meta即可</strong>。</p>
<figure class="highlight perl"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">spades.py --meta -<span class="number">1</span> read_1.fastq.gz -<span class="number">2</span> read_2.fastq.gz -o spades_result -t <span class="number">24</span></span><br></pre></td></tr></table></figure>

<h3 id="megahit"><a href="#megahit" class="headerlink" title="megahit"></a>megahit</h3><p>该软件运行速度快于SPAdes且占用的计算资源也比较少，<strong>但组装效果不如前者</strong>。</p>
<figure class="highlight perl"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">megahit -o megahit -<span class="number">1</span> read_1.fastq.gz -<span class="number">2</span> read_2.fastq.gz -t <span class="number">24</span></span><br></pre></td></tr></table></figure>

<h2 id="三代测序数据拼接"><a href="#三代测序数据拼接" class="headerlink" title="三代测序数据拼接"></a>三代测序数据拼接</h2><h3 id="flye"><a href="#flye" class="headerlink" title="flye"></a>flye</h3><p>质控后，三代测序宏基因组数据仍然可以使用flye进行组装，它其中有一个metaflye的模块专门用来组装宏基因组数据。</p>
<figure class="highlight perl"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">flye --nano-raw nano.fastq.gz -g <span class="number">5</span>m -t <span class="number">12</span> -o metaflye_result --meta</span><br></pre></td></tr></table></figure>

<p><strong>值得注意的是在进行完数据拼接后还可以使用纠错软件进行纠错（pilon、racon等），这里可能会涉及到使用二代测序数据辅助纠错</strong>。</p>
<h2 id="二代三代软件拼接结果对比"><a href="#二代三代软件拼接结果对比" class="headerlink" title="二代三代软件拼接结果对比"></a>二代三代软件拼接结果对比</h2><p><img src="https://raw.githubusercontent.com/Zhang-EK/blog_img/main/image.png"></p>
<p>这里可以很明显的看到同为二代测序拼接，megahit所拼接出的contigs不如SPAdes长。Nanopore三代测序在读长方面的优势还是比较明显的。</p>
<h2 id="宏基因组物种分类"><a href="#宏基因组物种分类" class="headerlink" title="宏基因组物种分类"></a>宏基因组物种分类</h2><h3 id="物种分类数据库"><a href="#物种分类数据库" class="headerlink" title="物种分类数据库"></a>物种分类数据库</h3><p>当前宏基因组物种分类的方法主要就是使用NCBI的物种分类数据库进行比对分类。将测序得到的序列和数据库（如nt库、nr库、refseq库）中的序列进行比对，如果两者具有很好的相似性则认为两者为同源序列，具有共同的祖先。</p>
<ul>
<li>nt库为Nucleotide Sequence Database，即核酸序列数据库，其中含有所有目前已知的核酸序列（基因组），是有冗余的。</li>
<li>nr库为Non-Redundant Protein Sequence Database，即非冗余蛋白库（只包含基因预测的蛋白而不是基因组）。一般DIAMOND比对使用这个数据库。</li>
<li>refseq库是NCBI官方验证的可以代表某个物种的参考序列所组成的数据库，是非冗余的，<strong>一般物种分类就是使用这个</strong>。</li>
<li>taxonomy数据库。这个库里的内容主要是物种的名字和种系，这些物种都至少在遗传数据库中有一条核酸或蛋白序列。其目的是为序列数据库建立一个一致的种系发生分类学，<strong>物种分类也需要这个</strong>。</li>
</ul>
<h3 id="物种鉴定软件"><a href="#物种鉴定软件" class="headerlink" title="物种鉴定软件"></a>物种鉴定软件</h3><img src="https://raw.githubusercontent.com/Zhang-EK/blog_img/main/111.png" style="zoom:67%;" />

<p>基本常用的软件都在上图中。一般DNA核酸比对还是使用<strong>Kraken2</strong>比较多，蛋白质一般使用<strong>DIAMOND</strong>和<strong>kauju</strong>，<strong>MetaPhlAn2</strong>也比较常用。</p>
<p><em><strong>可以使用mock数据集进行测试。</strong></em></p>
<h3 id="三代测序数据处理"><a href="#三代测序数据处理" class="headerlink" title="三代测序数据处理"></a>三代测序数据处理</h3><h4 id="Centrifuge"><a href="#Centrifuge" class="headerlink" title="Centrifuge"></a>Centrifuge</h4><p>这里主要介绍一下centrifuge软件，物种鉴定使用方式如下</p>
<figure class="highlight perl"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">centrifuge -<span class="keyword">x</span> centrifuge_h+p+v_20200318/hpv -U nanopore.fastq.gz --report-file report.tsv -S result.tsv -p <span class="number">64</span> &gt;centrifuge.log</span><br></pre></td></tr></table></figure>

<p>会输出两个结果</p>
<ol>
<li><p>按照reads进行结果统计的结果centrifuge_output.tsv</p>
<p><img src="https://raw.githubusercontent.com/Zhang-EK/blog_img/main/image2.png" alt="image2"></p>
</li>
</ol>
<p>一共是八列，从左到右分别是reads ID、比对上的序列的Accession ID、物种分类ID、序列比对评分、第二比对结果的评分、比对上的部分的序列长度、总reads长度、该reads比对上的物种序列个数。</p>
<ol start="2">
<li>按照比对上的物种进行的统计centrifuge_report.tsv</li>
</ol>
<p><img src="https://raw.githubusercontent.com/Zhang-EK/blog_img/main/image%20(1).png" alt="image (1)"></p>
<p>一共是七列，从左到右分别是比对上的物种名字、物种分类ID、物种层级、基因组大小、比对上的reads数量（每一个reads可以比对上多个物种）、唯一比对上该物种的reads数量、丰度（比对上的区域/整个基因组的长度）。</p>
<p>在得到了结果之后就可以使用R等方式对tsv表格进行过滤。比如一条reads会比对到数据库中的多个物种，通常我们需要选择每条reads最优的比对，并且还要去除掉比对上的reads太少的物种。当然，命令行也能进行结果筛选。</p>
<figure class="highlight perl"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">awk -F <span class="string">&quot;\t&quot;</span> <span class="string">&#x27;&#123;if ($3==&quot;species&quot; &amp;&amp; $6 &gt;5) print $1&quot;\t&quot;$6&#125;&#x27;</span> <span class="number">0</span>.<span class="number">01_</span>report.tsv &gt;<span class="number">0</span>.<span class="number">01</span>.txt</span><br></pre></td></tr></table></figure>

<p><em><strong>这里标记一篇文章，<a target="_blank" rel="noopener" href="https://www.nature.com/articles/s41587-019-0156-5">Nanopore metagenomics enables rapid clinical diagnosis of bacterial lower respiratory infection</a>，这篇文章中有40个潜在肺炎病人肺部宏基因组测序的结果，可以通过Centrifuge比对的方式得到样品中占决定性多数的细菌种类，如果该细菌与肺炎有关那么就判定该患者有较大可能患有肺炎。该文章证明利用纳米孔基因组测序和宏基因组分析可以快速地进行临床诊断下呼吸道感染。其中的数据可以用来练习宏基因组物种分类。</strong></em></p>
<h3 id="二代测序数据处理"><a href="#二代测序数据处理" class="headerlink" title="二代测序数据处理"></a>二代测序数据处理</h3><p>这一部分的几个软件或流程基本都来自于<strong>Biobakery workflow</strong>。这个流程来自于哈佛的Huttenhower团队。</p>
<img src="https://raw.githubusercontent.com/Zhang-EK/blog_img/main/32535.png" style="zoom:50%;" />

<ol>
<li>数据质控：使用Kneaddata，该软件先调用Trimmomatic过滤数据，然后利用bowtie2或bmtagger比对去除宿主数据。</li>
<li>如果不拼接数据，则可以直接使用MetaPhlAn2或者Kraken2实现序列的物种分类及物种丰度分析。功能鉴定则可以使用Humann完成。</li>
<li>也可以先对Reads进行拼接（Megahit、Metaspades等软件），然后使用Prokka软件从组装好的Contig或者Scafford预测基因，并使用cd-hit构建非冗余基因集。然后利用Nr、COG、GO、KEGG等数据库对基因进行注释。</li>
<li>如果有多样品则可以进行分组统计，将得到的物种/基因/功能表格利用R或者STAMP等软件在这三个层次进行分析。</li>
</ol>
<h4 id="Kneaddata"><a href="#Kneaddata" class="headerlink" title="Kneaddata"></a>Kneaddata</h4><p>Kneaddata是一个数据质控过滤流程，整合了Fasqc质控，Trimmomatic数据过滤，Bowtie2比对数据库过滤宿主等功能。输入原始数据后就可以得到Cleandata。默认集成了人、小鼠、rRNA等数据库。如果是其他宿主，可以自行建库。</p>
<figure class="highlight perl"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">kneaddata -i reads_1.fastq.gz -i reads_2.fastq.gz -db path\Homo_sapiens -o kneaddata_output --remove-intermediate-output -v -t <span class="number">12</span> --trimmomatic-options <span class="string">&#x27;ILLUMINACLIP:~/miniconda3/envs/biobakery/share/trimmomatic/adapters/TruSeq3-PE.fa:2:40:15 SLIDINGWINDOW:4:20 MINLEN:50&#x27;</span> --reorder --bowtie2-options <span class="string">&#x27;--very-sensitive --dovetail&#x27;</span> --run-fastqc-start --run-fastqc-end</span><br></pre></td></tr></table></figure>

<p><em><strong>数据库：/data/Food/primary/R0987_nextgen/Erkang.Zhang/20221130_course/course_4_meta/kneadData_database/</strong></em></p>
<h4 id="MetaPhlAn"><a href="#MetaPhlAn" class="headerlink" title="MetaPhlAn"></a>MetaPhlAn</h4><p>MetaPhlan是用于二代测序物种分类的工具，利用快速比对工具Bowtie2与Marker基因集比对（非全基因组），所以运行速度非常的快。输入Kneaddata过滤后的数据就可以快速得到微生物群体中的物种组成。</p>
<p>MetaPhlAn调用Bowtie2比对，可以输入多种格式fasta、fastq、bowtie2out、sam等。</p>
<figure class="highlight perl"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">metaphlan --input_type fastq --nproc <span class="number">12</span> --bowtie2out metagenome.bowtie2.bz2 reads_filter.<span class="number">1</span>.fastq.gz,reads_filter.<span class="number">2</span>.fastq.gz -o SRS011243_abundance_table.txt</span><br></pre></td></tr></table></figure>

<p>结果导出的表格分为两列，第一列为物种分类，按照层级从大到小排列。第二列是物种的相对丰度，每一层级加起来都是100%，有些软件输出的则是reads数目。</p>
<img src="https://raw.githubusercontent.com/Zhang-EK/blog_img/main/4375.png" alt="4375" style="zoom:50%;" />

<p><em><strong>数据库：/data/Food/primary/R0987_nextgen/Erkang.Zhang/20221130_course/course_4_meta/metaphlan_database/</strong></em></p>
<h4 id="HUMAnN"><a href="#HUMAnN" class="headerlink" title="HUMAnN"></a>HUMAnN</h4><p>HUMAnN不仅可以通过调用MetaPhlAn完成物种组成部分还可以准确地获得微生物代谢途径和功能模块信息。</p>
<figure class="highlight perl"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">humann --input-<span class="keyword">format</span> fastq --input reads_filter.<span class="number">1</span>.fastq.gz --input reads_filter.<span class="number">2</span>.fastq.gz --output result --threads <span class="number">12</span> --search-mode uniref9<span class="number">0</span></span><br></pre></td></tr></table></figure>

<p>结果会输出三个文件。</p>
<img src="https://raw.githubusercontent.com/Zhang-EK/blog_img/main/452415.png" alt="452415" style="zoom:67%;" />

<ol>
<li>基因家族文件（genefamilies）：群体中每个基因家族的丰度。基因家族是一组进化上相关的编码蛋白质序列，通常具有相似的功能。基因家族的丰度采用RPK表示（每kb的reads中该基因家族的含量），以此来标准化不同长度的基因家族的丰度。</li>
<li>通路丰度文件（pathabundance）：代表群体中通路的丰度。</li>
<li>通路覆盖文件（pathcoverage）</li>
</ol>
<p>另外，HUMAnM还支持比对到其他蛋白数据库的注释结果，以及对RPK的标准化/归一化处理，<a target="_blank" rel="noopener" href="https://zhuanlan.zhihu.com/p/240910229">见该网站</a>。</p>
<p><em><strong>数据库：/data/Food/primary/R0987_nextgen/Erkang.Zhang/20221130_course/course_4_meta/humann_database/</strong></em></p>
<h3 id="最终结果的可视化"><a href="#最终结果的可视化" class="headerlink" title="最终结果的可视化"></a>最终结果的可视化</h3><h4 id="Pavian可视化"><a href="#Pavian可视化" class="headerlink" title="Pavian可视化"></a>Pavian可视化</h4><p>Pavian是一款R包，<a target="_blank" rel="noopener" href="https://fbreitwieser.shinyapps.io/pavian/">但也有网页形式</a>。可以用来可视化Kraken、MetaPhlAn、Centrifuge等软件的结果。如果要可视化Centrifuge的结果，需要先将其转化为Kraken的结果格式。</p>
<figure class="highlight perl"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">centrifuge-kreport -<span class="keyword">x</span> centrifuge_p/p_compressed result_log.tsv &gt;kraken_result_log.tsv</span><br></pre></td></tr></table></figure>

<p>然后将其上传到网页上即可，可以绘制桑基图等。</p>
<h4 id="Megan-可视化"><a href="#Megan-可视化" class="headerlink" title="Megan 可视化"></a>Megan 可视化</h4><p>Megan是一款综合性的微生物物种分类工具，除了支持物种分类还支持功能分析（KEGG数据库使用的是免费版本，付费版Megan可以使用全数据库）。支持多种格式输入，diamond比对的daa格式、blast比对格式、sam、biom等等。也可以用最简单的tsv、csv，两列即可，第一列物种名称、第二列物种丰度信息或reads数目。</p>
<h4 id="GraPhlAn"><a href="#GraPhlAn" class="headerlink" title="GraPhlAn"></a>GraPhlAn</h4><p>也在Biobakery中，输入MetaPhlAn的结果可以直接绘制。</p>
<h2 id="注意事项"><a href="#注意事项" class="headerlink" title="注意事项"></a>注意事项</h2><ul>
<li>现在的宏基因组分析流程基本都是<strong>先拼接生成Contig</strong>再继续分析物种、基因。</li>
</ul>

    </div>

    
    
    

      <footer class="post-footer">
          <div class="post-tags">
              <a href="/tags/%E7%94%9F%E4%BF%A1%E8%BD%AF%E4%BB%B6%E5%AD%A6%E4%B9%A0%E7%AC%94%E8%AE%B0/" rel="tag"># 生信软件学习笔记</a>
          </div>

        


        
    <div class="post-nav">
      <div class="post-nav-item">
    <a href="/2022/12/09/%E5%9F%BA%E5%9B%A0%E6%B5%8B%E5%BA%8F%E5%88%86%E6%9E%90%E8%BD%AF%E4%BB%B6%E5%8F%8A%E6%B5%81%E7%A8%8B%EF%BC%88%E9%9D%9Emeta%EF%BC%89/" rel="prev" title="二代、三代测序软件及分析流程整理">
      <i class="fa fa-chevron-left"></i> 二代、三代测序软件及分析流程整理
    </a></div>
      <div class="post-nav-item">
    <a href="/2023/01/09/Alpha%E5%92%8CBeta%E5%A4%9A%E6%A0%B7%E6%80%A7/" rel="next" title="多样性指数">
      多样性指数 <i class="fa fa-chevron-right"></i>
    </a></div>
    </div>
      </footer>
    
  </article>
  
  
  



          </div>
          

<script>
  window.addEventListener('tabs:register', () => {
    let { activeClass } = CONFIG.comments;
    if (CONFIG.comments.storage) {
      activeClass = localStorage.getItem('comments_active') || activeClass;
    }
    if (activeClass) {
      let activeTab = document.querySelector(`a[href="#comment-${activeClass}"]`);
      if (activeTab) {
        activeTab.click();
      }
    }
  });
  if (CONFIG.comments.storage) {
    window.addEventListener('tabs:click', event => {
      if (!event.target.matches('.tabs-comment .tab-content .tab-pane')) return;
      let commentClass = event.target.classList[1];
      localStorage.setItem('comments_active', commentClass);
    });
  }
</script>

        </div>
          
  
  <div class="toggle sidebar-toggle">
    <span class="toggle-line toggle-line-first"></span>
    <span class="toggle-line toggle-line-middle"></span>
    <span class="toggle-line toggle-line-last"></span>
  </div>

  <aside class="sidebar">
    <div class="sidebar-inner">

      <ul class="sidebar-nav motion-element">
        <li class="sidebar-nav-toc">
          Table of Contents
        </li>
        <li class="sidebar-nav-overview">
          Overview
        </li>
      </ul>

      <!--noindex-->
      <div class="post-toc-wrap sidebar-panel">
          <div class="post-toc motion-element"><ol class="nav"><li class="nav-item nav-level-1"><a class="nav-link" href="#%E5%AE%8F%E5%9F%BA%E5%9B%A0%E7%BB%84%E6%B5%8B%E5%BA%8F%E5%8F%8A%E5%88%86%E6%9E%90%E6%B5%81%E7%A8%8B"><span class="nav-number">1.</span> <span class="nav-text">宏基因组测序及分析流程</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#%E4%BA%8C%E4%BB%A3%E6%B5%8B%E5%BA%8F%E6%95%B0%E6%8D%AE%E6%8B%BC%E6%8E%A5"><span class="nav-number">1.1.</span> <span class="nav-text">二代测序数据拼接</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#SPAdes"><span class="nav-number">1.1.1.</span> <span class="nav-text">SPAdes</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#megahit"><span class="nav-number">1.1.2.</span> <span class="nav-text">megahit</span></a></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#%E4%B8%89%E4%BB%A3%E6%B5%8B%E5%BA%8F%E6%95%B0%E6%8D%AE%E6%8B%BC%E6%8E%A5"><span class="nav-number">1.2.</span> <span class="nav-text">三代测序数据拼接</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#flye"><span class="nav-number">1.2.1.</span> <span class="nav-text">flye</span></a></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#%E4%BA%8C%E4%BB%A3%E4%B8%89%E4%BB%A3%E8%BD%AF%E4%BB%B6%E6%8B%BC%E6%8E%A5%E7%BB%93%E6%9E%9C%E5%AF%B9%E6%AF%94"><span class="nav-number">1.3.</span> <span class="nav-text">二代三代软件拼接结果对比</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#%E5%AE%8F%E5%9F%BA%E5%9B%A0%E7%BB%84%E7%89%A9%E7%A7%8D%E5%88%86%E7%B1%BB"><span class="nav-number">1.4.</span> <span class="nav-text">宏基因组物种分类</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#%E7%89%A9%E7%A7%8D%E5%88%86%E7%B1%BB%E6%95%B0%E6%8D%AE%E5%BA%93"><span class="nav-number">1.4.1.</span> <span class="nav-text">物种分类数据库</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#%E7%89%A9%E7%A7%8D%E9%89%B4%E5%AE%9A%E8%BD%AF%E4%BB%B6"><span class="nav-number">1.4.2.</span> <span class="nav-text">物种鉴定软件</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#%E4%B8%89%E4%BB%A3%E6%B5%8B%E5%BA%8F%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86"><span class="nav-number">1.4.3.</span> <span class="nav-text">三代测序数据处理</span></a><ol class="nav-child"><li class="nav-item nav-level-4"><a class="nav-link" href="#Centrifuge"><span class="nav-number">1.4.3.1.</span> <span class="nav-text">Centrifuge</span></a></li></ol></li><li class="nav-item nav-level-3"><a class="nav-link" href="#%E4%BA%8C%E4%BB%A3%E6%B5%8B%E5%BA%8F%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86"><span class="nav-number">1.4.4.</span> <span class="nav-text">二代测序数据处理</span></a><ol class="nav-child"><li class="nav-item nav-level-4"><a class="nav-link" href="#Kneaddata"><span class="nav-number">1.4.4.1.</span> <span class="nav-text">Kneaddata</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#MetaPhlAn"><span class="nav-number">1.4.4.2.</span> <span class="nav-text">MetaPhlAn</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#HUMAnN"><span class="nav-number">1.4.4.3.</span> <span class="nav-text">HUMAnN</span></a></li></ol></li><li class="nav-item nav-level-3"><a class="nav-link" href="#%E6%9C%80%E7%BB%88%E7%BB%93%E6%9E%9C%E7%9A%84%E5%8F%AF%E8%A7%86%E5%8C%96"><span class="nav-number">1.4.5.</span> <span class="nav-text">最终结果的可视化</span></a><ol class="nav-child"><li class="nav-item nav-level-4"><a class="nav-link" href="#Pavian%E5%8F%AF%E8%A7%86%E5%8C%96"><span class="nav-number">1.4.5.1.</span> <span class="nav-text">Pavian可视化</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#Megan-%E5%8F%AF%E8%A7%86%E5%8C%96"><span class="nav-number">1.4.5.2.</span> <span class="nav-text">Megan 可视化</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#GraPhlAn"><span class="nav-number">1.4.5.3.</span> <span class="nav-text">GraPhlAn</span></a></li></ol></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#%E6%B3%A8%E6%84%8F%E4%BA%8B%E9%A1%B9"><span class="nav-number">1.5.</span> <span class="nav-text">注意事项</span></a></li></ol></li></ol></div>
      </div>
      <!--/noindex-->

      <div class="site-overview-wrap sidebar-panel">
        <div class="site-author motion-element" itemprop="author" itemscope itemtype="http://schema.org/Person">
    <img class="site-author-image" itemprop="image" alt="ZEK YL"
      src="https://raw.githubusercontent.com/Zhang-EK/blog_img/main/WechatIMG22.jpeg">
  <p class="site-author-name" itemprop="name">ZEK YL</p>
  <div class="site-description" itemprop="description">2019.10.16</div>
</div>
<div class="site-state-wrap motion-element">
  <nav class="site-state">
      <div class="site-state-item site-state-posts">
          <a href="/archives">
          <span class="site-state-item-count">9</span>
          <span class="site-state-item-name">posts</span>
        </a>
      </div>
      <div class="site-state-item site-state-categories">
            <a href="/categories/">
        <span class="site-state-item-count">2</span>
        <span class="site-state-item-name">categories</span></a>
      </div>
      <div class="site-state-item site-state-tags">
            <a href="/tags/">
        <span class="site-state-item-count">3</span>
        <span class="site-state-item-name">tags</span></a>
      </div>
  </nav>
</div>
  <div class="links-of-author motion-element">
      <span class="links-of-author-item">
        <a href="https://github.com/Zhang-EK" title="GitHub → https:&#x2F;&#x2F;github.com&#x2F;Zhang-EK" rel="noopener" target="_blank"><i class="fab fa-github fa-fw"></i></a>
      </span>
      <span class="links-of-author-item">
        <a href="mailto:zhangerkang73@gmail.com" title="E-Mail → mailto:zhangerkang73@gmail.com" rel="noopener" target="_blank"><i class="fa fa-envelope fa-fw"></i></a>
      </span>
      <span class="links-of-author-item">
        <a href="https://scholar.google.com.hk/citations?user=e3kZB8YAAAAJ&hl=zh-CN" title="Google学术 → https:&#x2F;&#x2F;scholar.google.com.hk&#x2F;citations?user&#x3D;e3kZB8YAAAAJ&amp;hl&#x3D;zh-CN" rel="noopener" target="_blank"><i class="fab fa-google fa-fw"></i></a>
      </span>
  </div>



      </div>

    </div>
  </aside>
  <div id="sidebar-dimmer"></div>


      </div>
    </main>

    <footer class="footer">
      <div class="footer-inner">
        

        

<div class="copyright">
  
  &copy; 
  <span itemprop="copyrightYear">2023</span>
  <span class="with-love">
    <i class="fa fa-heart"></i>
  </span>
  <span class="author" itemprop="copyrightHolder">ZEK YL</span>
</div>

        








      </div>
    </footer>
  </div>

  
  <script src="/lib/anime.min.js"></script>
  <script src="/lib/velocity/velocity.min.js"></script>
  <script src="/lib/velocity/velocity.ui.min.js"></script>

<script src="/js/utils.js"></script>

<script src="/js/motion.js"></script>


<script src="/js/schemes/pisces.js"></script>


<script src="/js/next-boot.js"></script>




  















  

  

  

</body>
</html>
